src/agent/transform-headers-agent.js

```diff
@@ -24,7 +24,11 @@ class TransformHeadersAgent extends WrappedAgent {
         const keys = request.getHeaderNames();
 
         for (const key of keys) {
-            headers[this.toPascalCase(key)] = request.getHeader(key);
+            if (key.toLowerCase().startsWith('x-')) {
+                headers[key] = request.getHeader(key);
+            } else {
+                headers[this.toPascalCase(key)] = request.getHeader(key);
+            }
```

Comment:
Oh, sorry for the confusion. The `x-something` was just an example. The header could be `my-header` as well or some other randomness. We've seen quite a few.

Also I think there are some `X-` headers which are actually sent by the browsers like `X-Requested-With`.

So the determination of the "custom headers" is a bit more complex.

Reply:
Lemme patch this real quick

Reply:
`X-Requested-With` doesn't exist in the spec: https://www.iana.org/assignments/message-headers/message-headers.xhtml nor in the MDN docs: https://developer.mozilla.org/pl/docs/Web/HTTP/Headers

Reply:
Yeah, the generator should always return correct casing. Unless we encounter some crap UA like the `applebot`.

Reply:
And yeah, I might be wrong about the `X-Requested-With`. I've seen it multiple times so I thought it's sent by the browser but maybe it's just common among developers to include it.

<img width="566" alt="image" src="https://user-images.githubusercontent.com/23726914/128166015-b632a5a4-c241-48ff-aeb1-b886895a42b6.png">

Reply:
Then this PR is good to go I think. I'm planning to optimize HTTP/2 related stuff next such as the ALPN negotiation. I think there's no need to manually store the cache anymore.